org.eclipse.scanning.device.ui/src/org/eclipse/scanning/device/ui/points/ValidateResultsView.java

```diff
@@ -0,0 +1,239 @@
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.ui.ISelectionListener;
+import org.eclipse.ui.IWorkbenchPart;
+import org.eclipse.ui.part.ViewPart;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * 
+ * A view which attempts to pick up selection events
+ * from calling validate on a device, and display the
+ * results of the validation to a user
+ * 
+ * @author Matt Taylor
+ *
+ */
+public class ValidateResultsView extends ViewPart implements ISelectionListener {
+
+	public static final String ID = "org.eclipse.scanning.device.ui.scan.validateResultsView"; //$NON-NLS-1$
+	private static final Logger logger = LoggerFactory.getLogger(ValidateResultsView.class);
+	
+	// UI
+	private StyledText text;
+	
+	public ValidateResultsView() {
+		PageUtil.getPage(getSite()).addSelectionListener(this);
+	}
+
+	/**
+	 * Create contents of the view part.
+	 * @param parent
+	 */
+	@Override
+	public void createPartControl(Composite parent) {
+		
+		Composite container = new Composite(parent, SWT.NONE);
+		container.setLayout(new GridLayout(1, false));
+		
+		this.text = new StyledText(container, SWT.MULTI | SWT.H_SCROLL | SWT.V_SCROLL);
+		text.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+		text.setBackground(text.getDisplay().getSystemColor(SWT.COLOR_WHITE));
+		text.getParent().layout(new Control[]{text});
+		
+		// We force the scan view to exist. It might be the one to return the compound model
+		// that we will use.
+		ScanningPerspective.createKeyPlayers();
+		
+        createActions();
+	}
+	
+	/**
+	 * Create the actions
+	 */
+	private void createActions() {
+		final IContributionManager man = getViewSite().getActionBars().getToolBarManager();
+
+		final Action restart = new Action("Clear", Activator.getImageDescriptor("icons/layers-stack.png")) {
+			public void run() {
+				clear();
+			}
+		};
+		man.add(restart);
+
+		final MenuManager menuMan = new MenuManager();
+		menuMan.add(restart);
+		
+		text.setMenu(menuMan.createContextMenu(text));
+	}
+
+	@Override
+	public void dispose() {
+		if (PageUtil.getPage(getSite())!=null) PageUtil.getPage(getSite()).removeSelectionListener(this);
+		super.dispose();
+	}
+	
+	private void clear() {
+		setThreadSafeText(text, new StyledString(""));
+	}
+	
+	@Override
+	public void selectionChanged(IWorkbenchPart part, ISelection selection) {
+		if (selection instanceof IStructuredSelection) {
+			Object ob = ((IStructuredSelection)selection).getFirstElement();
+
+			if (ob instanceof ValidateResults) {
+				ValidateResults validationResults  = (ValidateResults)ob;
+				update(validationResults);
+			}
+
+		}
+	}
+
+	/**
+	 * Method for updating the display with the validation results
+	 * @param results the ValidationResults object
+	 */
+	public void update(ValidateResults results) {
+		if (results == null) return;

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)